python/ray/_raylet.pyx

```diff
@@ -1722,6 +1727,21 @@ cdef class CoreWorker:
             self.job_config.ParseFromString(c_job_config.SerializeAsString())
         return self.job_config
 
+    def prepare_runtime_env(self, serialized_runtime_env: str) -> str:
+        """Update parent's runtime env with new env via a simple dict update.
+
+        If the resulting runtime env is empty, fall back to the runtime env
+        set in the JobConfig.
+        """
+        runtime_env_dict = json.loads(self.current_serialized_runtime_env()
+                                      or "{}")
```

Comment:
Is there a reason we store this in the serialized format? If we are just going to deserialize it every time we use it it'd be better to just store it as a dict

Reply:
I think it's because it grabs the current (parent) runtime env from the WorkerContext in C++, where we don't want to just store it as a dict.  But maybe there's a corresponding WorkerContext-like thing in Python?  I'll take a closer look, but let me know if you know offhand

Reply:
given that it's static you should be able to just cache it in the `CoreWorker` cython class